src/language-js/printer-estree.js

```diff
@@ -120,6 +120,16 @@ const {
 let uid = 0;
 
 function shouldPrintComma(options, level) {
+  // angular does not allow trailing comma
+  if (
+    options.parser === "__ng_interpolation" ||
+    options.parser === "__ng_action" ||
+    options.parser === "__ng_binding" ||
+    options.parser === "__ng_directive"
+  ) {
```

Comment:
Can we simplify this using regexp `^__ng_`?

Reply:
really nessary? only 4, but I was thinking override here, https://github.com/prettier/prettier/blob/4ed525350904a3c19f4a7504e53a27a98b12531e/src/language-js/parser-angular.js#L5 what do you suggest?

Reply:
Just realize this options is not passing to print function

```js
options.trailingComma = "none";
```

seems not safe. maybe I should keep check in shouldPrintComma

Reply:
@evilebottnawi still prefer regexp?

Reply:
It is just question, i am not familiar with angular very good,all fine